scripts/smoothwall/HandleDansGuardianLog.pl

```diff
@@ -59,44 +93,52 @@ sub fetch_relevant_url_categories {
 	foreach $category (values $categories->{'categories'}) {
 		$relevant_url_categories{$category} .= "";
 	}
+	foreach $category (keys %relevant_url_categories) {
+		log_info "Loaded category $category";
+	}
+	$relevant_url_categories_load_time = time();
+	log_info "Finished loading relevant categories";
 }
 
 sub handle_records_from_stream {
-	my ($fh, $analysis_engine_url) = @_;
+	my ($fh) = @_;
 
-	fetch_relevant_url_categories $analysis_engine_url;
+	fetch_relevant_url_categories; # To check the connection, before receiving the first event
 	while (<$fh>) {
+		fetch_relevant_url_categories
 		my $analysis_event_json = transform_log_record $_;
 
 		if ($analysis_event_json) {
-			print Dumper $analysis_event_json;
 			my $post_result = $ua->request(POST $analysis_engine_url, Content_Type => 'application/json', Content => $analysis_event_json);
 			my $status_code = $post_result->{'_rc'};
 			if ($status_code != 200) {
-				print STDERR "ERROR: POST to '$analysis_engine_url' returned status $status_code\n";
+				log_error "POST to '$analysis_engine_url' returned status $status_code";
 			}
 		}
 	}
 }
 
 sub handle_records_from_file {
-	my ($file_name, $analysis_engine_url) = @_;
+	my ($file_name) = @_;
+	log_info "Reading records from '$file_name'";
 	open INPUT_FILE, "<$file_name" or die "Cannot open file $INPUT_FILE\n";
-	handle_records_from_stream(*INPUT_FILE, $analysis_engine_url);
+	handle_records_from_stream(*INPUT_FILE);
 	close INPUT_FILE;
 }
 
-my $analysis_engine_url = 'http://localhost:8081/analysisEngine/';
-GetOptions ('analysisEngineURL=s' => \$analysis_engine_url) or die "Usage: $0 [--analysisEngineURL <URL>] [<input file>]";
+$| = 1; # Make STDOUT unbuffered
+GetOptions ('analysisEngineURL=s' => \$analysis_engine_url,
+	'categoriesRefreshInterval=i' => \$categories_refresh_interval) or die "Usage: $0 [--analysisEngineURL <URL>] [<input file>]";
```

Comment:
I see. It can be added to this line:

    or die "Usage: $0 [--analysisEngineURL <URL>] [--categoriesRefreshInterval <seconds integer>] [<input file>]"